### PR TITLE
best-practice: support extra strategy for fine-tuning the performance in massive regions

### PR DESCRIPTION
**Pull request by LykxSassinator**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.2 (TiDB 8.2 versions)
- [x] v8.1 (TiDB 8.1 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [x] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by Connor1996 — APPROVED** [review]

File: best-practices/massive-regions-best-practices.md [hunk 1]
```diff
@@ -51,6 +51,14 @@ aliases: ['/docs-cn/dev/best-practices/massive-regions-best-practices/','/docs-c
 
     ![图 3 查看 Propose wait duration](/media/best-practices/propose-wait-duration.png)
 
++ Raft IO 下的 `Commit log duration`
+
+    `Commit log duration` 是 Raftstore 提交 Raft 日志到相应 Region 的多数派成员所花费的时间。如果该延迟时间较长且存在大幅度波动，说明 Raftstore 比较繁忙或者处理 append log 比较慢，又或者是由于网络阻塞导致不能及时提交 Raft 日志。
```
Comment (qiancai) [inline best-practices/massive-regions-best-practices.md:56]:
```suggestion
    `Commit log duration` 是 Raftstore 将 Raft 日志提交到相应 Region 的多数成员所花费的时间。如果该指标的值较高且波动较大，说明 Raftstore 的负载较高，或者 append log 较慢，又或者 Raft 日志由于网络阻塞无法及时提交。
```

File: tikv-configuration-file.md [hunk 2]
```diff
@@ -165,7 +165,7 @@ TiKV 配置文件比命令行参数支持更多的选项。你可以在 [etc/con
 
 ### `grpc-raft-conn-num`
 
-+ TiKV 节点之间用于 Raft 通讯的链接最大数量。
++ TiKV 节点之间用于 Raft 通讯的连接最大数量。
```
Comment (qiancai) [inline tikv-configuration-file.md:168]:
```suggestion
+ TiKV 节点之间用于 Raft 通信的连接最大数量。
```

File: best-practices/massive-regions-best-practices.md [hunk 3]
```diff
@@ -132,6 +140,14 @@ Region 默认的大小约为 96 MiB，将其调大也可以减少 Region 个数
 > + 查询性能回退，尤其是大范围数据查询的性能会有回退。
 > + 调度变慢。
 
+### 方法七：提高 Raft 通信连接数
+
+Raft 通信的默认连接数为 1，将其调大可以减少因为海量 Region 通信量过大导致的阻塞情况。配置的具体说明可以参考[grpc-raft-conn-num](/tikv-configuration-file.md#grpc-raft-conn-num)
```
Comment (qiancai) [inline best-practices/massive-regions-best-practices.md:145]:
```suggestion
TiKV 节点间用于 Raft 通信的连接数默认为 1，将其调大可以减少因为海量 Region 通信量过大而导致的阻塞情况。具体的配置说明可以参考 [`grpc-raft-conn-num`](/tikv-configuration-file.md#grpc-raft-conn-num)。
```

File: best-practices/massive-regions-best-practices.md [hunk 4]
```diff
@@ -132,6 +140,14 @@ Region 默认的大小约为 96 MiB，将其调大也可以减少 Region 个数
...
+
+> **注意：**
+>
+> 连接数不宜过大，建议范围为 [1, 4]
```
Comment (qiancai) [inline best-practices/massive-regions-best-practices.md:149]:
```suggestion
> 为了 xxx，连接数不宜设置过大，建议范围为 `[1, 4]`。
```

Comment (qiancai) [reply best-practices/massive-regions-best-practices.md:149]: xxx 需要替换为原因

File: best-practices/massive-regions-best-practices.md [hunk 5]
```diff
@@ -132,6 +140,14 @@ Region 默认的大小约为 96 MiB，将其调大也可以减少 Region 个数
 > + 查询性能回退，尤其是大范围数据查询的性能会有回退。
 > + 调度变慢。
 
+### 方法七：提高 Raft 通信连接数
+
+TiKV 节点间用于 Raft 通信的连接数默认为 1，将其调大可以减少因为海量 Region 通信量过大而导致的阻塞情况。具体的配置说明可以参考 [`grpc-raft-conn-num`](/tikv-configuration-file.md#grpc-raft-conn-num)。
+
+> **注意：**
+>
+> 为了减少非必要的线程切换开销以及批量化效果差的影响，建议范围为 `[1, 4]`
```
Comment (qiancai) [inline best-practices/massive-regions-best-practices.md:149]:
```suggestion
> 为了减少不必要的线程切换开销以及批量处理效果不佳的影响，建议范围为 `[1, 4]`
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

File: best-practices/massive-regions-best-practices.md [hunk 6]
```diff
@@ -51,6 +51,14 @@ aliases: ['/docs-cn/dev/best-practices/massive-regions-best-practices/','/docs-c
 
     ![图 3 查看 Propose wait duration](/media/best-practices/propose-wait-duration.png)
 
++ Raft IO 下的 `Commit log duration`
+
+    `Commit log duration` 是 Raftstore 将 Raft 日志提交到相应 Region 的多数成员所花费的时间。如果该指标的值较高且波动较大，说明 Raftstore 的负载较高，或者 append log 较慢，又或者 Raft 日志由于网络阻塞无法及时提交。
```
Comment (hfxsd) [inline best-practices/massive-regions-best-practices.md:56]:
```suggestion
    `Commit log duration` 是 Raftstore 将 Raft 日志提交到相应 Region 的多数成员所花费的时间。如果该指标的值较高且波动较大，可能的原因有：
    
    - Raftstore 的负载较高
    - append log 较慢
    - Raft 日志由于网络阻塞无法及时提交
```

File: best-practices/massive-regions-best-practices.md [hunk 7]
```diff
@@ -132,6 +140,14 @@ Region 默认的大小约为 96 MiB，将其调大也可以减少 Region 个数
 > + 查询性能回退，尤其是大范围数据查询的性能会有回退。
 > + 调度变慢。
 
+### 方法七：提高 Raft 通信连接数
```
Comment (hfxsd) [inline best-practices/massive-regions-best-practices.md:143]:
```suggestion
### 方法七：提高 Raft 通信的连接最大数量
```

**Review by hfxsd — APPROVED** [review]

File: best-practices/massive-regions-best-practices.md [hunk 8]
```diff
@@ -132,6 +140,14 @@ Region 默认的大小约为 96 MiB，将其调大也可以减少 Region 个数
...
+
+TiKV 节点间用于 Raft 通信的连接数默认为 1，将其调大可以减少因为海量 Region 通信量过大而导致的阻塞情况。具体的配置说明可以参考 [`grpc-raft-conn-num`](/tikv-configuration-file.md#grpc-raft-conn-num)。
```
Comment (hfxsd) [inline best-practices/massive-regions-best-practices.md:145]:
```suggestion
TiKV 节点间用于 Raft 通信的最大连接数默认为 1，将其调大可以减少因为海量 Region 通信量过大而导致的阻塞情况。具体的配置说明可以参考 [`grpc-raft-conn-num`](/tikv-configuration-file.md#grpc-raft-conn-num)。
```

File: best-practices/massive-regions-best-practices.md [hunk 9]
```diff
@@ -51,6 +51,18 @@ aliases: ['/docs-cn/dev/best-practices/massive-regions-best-practices/','/docs-c
...
 
++ Raft IO 下的 `Commit log duration`
+
+    `Commit log duration` 是 Raftstore 将 Raft 日志提交到相应 Region 的多数成员所花费的时间。如果该指标的值较高且波动较大，可能的原因有：
+    
+    - Raftstore 的负载较高
+    - append log 较慢
+    - Raft 日志由于网络阻塞无法及时提交
+
+    参考值：低于 200-500ms。
```
Comment (qiancai) [inline best-practices/massive-regions-best-practices.md:62]:
```suggestion
  参考值：低于 200-500ms。
```

**Pull request merged**